invoice/src/main/java/org/killbill/billing/invoice/generator/InvoiceDateUtils.java

```diff
@@ -69,23 +71,38 @@ public static BigDecimal calculateProRationAfterLastBillingCycleDate(final Local
      * @param endDate                  end date of the prorated interval
      * @param previousBillingCycleDate start date of the period
      * @param nextBillingCycleDate     end date of the period
+     * @param fixedDaysInMonth         fixed days to consider in a month (to avoid proration)
      */
-    private static BigDecimal calculateProrationBetweenDates(final LocalDate startDate, final LocalDate endDate, final LocalDate previousBillingCycleDate, final LocalDate nextBillingCycleDate) {
-        final int daysBetween = Days.daysBetween(previousBillingCycleDate, nextBillingCycleDate).getDays();
-        return calculateProrationBetweenDates(startDate, endDate, daysBetween);
+    private static BigDecimal calculateProrationBetweenDates(final LocalDate startDate, final LocalDate endDate, final LocalDate previousBillingCycleDate, final LocalDate nextBillingCycleDate, final int fixedDaysInMonth) {
+        final int daysBetween = fixedDaysInMonth == 0 ? Days.daysBetween(previousBillingCycleDate, nextBillingCycleDate).getDays() : fixedDaysInMonth;
+        return calculateProrationBetweenDates(startDate, endDate, daysBetween, fixedDaysInMonth);
     }
 
-    public static BigDecimal calculateProrationBetweenDates(final LocalDate startDate, final LocalDate endDate, int daysBetween) {
+    public static BigDecimal calculateProrationBetweenDates(final LocalDate startDate, final LocalDate endDate, final int daysBetween, final int fixedDaysInMonth) {
         if (daysBetween <= 0) {
             return BigDecimal.ZERO;
         }
 
         final BigDecimal daysInPeriod = new BigDecimal(daysBetween);
-        final BigDecimal days = new BigDecimal(Days.daysBetween(startDate, endDate).getDays());
+        final BigDecimal days = fixedDaysInMonth == 0 ? new BigDecimal(Days.daysBetween(startDate, endDate).getDays()) : new BigDecimal(daysBetweenWithFixedDaysInMonth(startDate, endDate, fixedDaysInMonth));
 
         return days.divide(daysInPeriod, KillBillMoney.MAX_SCALE, KillBillMoney.ROUNDING_METHOD);
     }
 
+    @VisibleForTesting
+    public static int daysBetweenWithFixedDaysInMonth(final LocalDate startDate, final LocalDate endDate, final int fixedDaysInMonth) {
+        final int daysBetween = Days.daysBetween(startDate, endDate).getDays();
+        if(startDate.getMonthOfYear() == endDate.getMonthOfYear()) { //same month, no need for extra logic
+            return daysBetween;
+        }
+        final int lastDayOfMonth = startDate.dayOfMonth().withMaximumValue().getDayOfMonth();
+        if (lastDayOfMonth == fixedDaysInMonth) {
+            return daysBetween;
+        } else {
+            return daysBetween - (lastDayOfMonth - fixedDaysInMonth);
```

Comment:
It's 🎉  that it works for both when `lastDayOfMonth > fixedDaysInMonth` or the reverse (and even the equality has commented above).